nupic/research/frameworks/dendrites/mixins/eval_per_task.py

```diff
@@ -0,0 +1,101 @@
+#  Numenta Platform for Intelligent Computing (NuPIC)
+#  Copyright (C) 2021, Numenta, Inc.  Unless you have an agreement
+#  with Numenta, Inc., for a separate license for this software code, the
+#  following terms and conditions apply:
+#
+#  This program is free software you can redistribute it and/or modify
+#  it under the terms of the GNU Affero Public License version 3 as
+#  published by the Free Software Foundation.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#  See the GNU Affero Public License for more details.
+#
+#  You should have received a copy of the GNU Affero Public License
+#  along with this program.  If not, see htt"://www.gnu.org/licenses.
+#
+#  http://numenta.org/licenses/
+#
+
+__all__ = [
+    "EvalPerTask",
+]
+
+
+class EvalPerTask:
+    """
+    Mixin for getting metrics, namely accuracy, for each task independently
+    """
+
+    def validate(self, loader=None):
+        """
+        Run validation on the currently active tasks, and gather accuracy measures
+        for each of those tasks independently.
+        """
+
+        # Figure this out
+        if loader is None:
+            loader = self.val_loader
+
+        results = {}
+        for task in range(self.current_task + 1):
+            self.val_loader.sampler.set_active_tasks(task)
+
+            # Store results for each individual task in results[task]
+            task_results = super().validate(loader)
```

Comment:
This mixin isn't allowing `validate` to be called with the validation loader's active tasks set to all observed tasks, because you're overwriting the `validate` method. How about you prepend `results = super().validate(loader)` at the beginning of this function so that the exeriment can also validate all tasks in aggregate?

Reply:
I might be misunderstanding, but averaging at the end is intended to accomplish the same thing as evaluating in aggregate. Evaluating in aggregate and then evaluating per task seems like redundant computation.

Reply:
Ok cool.